### ISSUE TITLE
Installing Pi-Hole - Could not update local repository

### ISSUE THREAD

**Issue by MrAdrianR** [pi-hole/pi-hole#1919]
**In raising this issue, I confirm the following:**

- [x] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [X] The issue I am reporting can be *replicated*.
- [X] The issue I am reporting isn't a duplicate (see [FAQs](https://github.com/pi-hole/pi-hole/wiki/FAQs), [closed issues](https://github.com/pi-hole/pi-hole/issues?utf8=%E2%9C%93&q=is%3Aissue%20is%3Aclosed%20), and [open issues](https://github.com/pi-hole/pi-hole/issues)).

**How familiar are you with the the source code relevant to this issue?:**

3

---
**Expected behaviour:**

First of all: Yes I read contributors guide, with my really bad English skills. (So sorry for that :( , And maybe its a duplicate issue. Sorry for that either.)
And yes, i already installed a couple of times pi-hole.
I actually know how the "Back-End product" should look like.
But after reinstalling Ubuntu Mate on my Rasp3b, and doing the basic things like: apt-get update/upgrade, and running the script `curl -sSL https://install.pi-hole.net | bash` as root user ill get after some checkboxes the `Error: Could not update local repository. Contact support.`

**Actual behaviour:**

` [✓] Root user check
  [✓] Disk space check

  [✓] Update local cache of available packages

  [✓] Checking apt-get for upgraded packages... up to date!

  [i] Installer Dependency checks...
  [✓] Checking for apt-utils
  [✓] Checking for dialog
  [✓] Checking for debconf
  [✓] Checking for dhcpcd5
  [✓] Checking for git
  [✓] Checking for iproute2
  [✓] Checking for whiptail

  [✓] Stopping dnsmasq service...

  [✓] Stopping lighttpd service...
  [i] Using interface: enxb827ebda9ba9
  [i] Using Google DNS servers
  [i] Static IP already configured
  [i] Unable to find IPv6 ULA/GUA address, IPv6 adblocking will not be enabled
  [i] IPv4 address: (IP Adress)
  [i] IPv6 address:
  [i] Web Interface On
  [i] Logging On.
  [✗] Check for existing repository in /etc/.pihole
  [i] Clone https://github.com/pi-hole/pi-hole.git into /etc/.pihole...
  Error: Could not update local repository. Contact support.`

**Steps to reproduce:**

curl -sSL https://install.pi-hole.net | bash
accepted all settings with "recommanded"

**Debug token provided by [uploading `pihole -d` log](https://discourse.pi-hole.net/t/the-pihole-command-with-examples/738#debug):**

no token

**Troubleshooting undertaken, and/or other relevant information:**
Maybe nice to know: i already installed tightvnc, but i think that is not the problem there.

I'll already searched for some reddit post and looked up of the same problem in your issue category. But sadly, i did not found anything that helped me a lot. I tried to solve the problem with these commands:
rm -R /var/www/html/admin
git clone https://github.com/pi-hole/AdminLTE /var/www/html/admin
pihole -r

rm -rf /var/www/html/admin
git clone https://github.com/pi-hole/adminLTE.git /var/www/html/admin

I am very sorry for this comment, my bad english skills and a duplication of an issue!

**Comment by Mcat12** [comment]
What is the output of those `git clone` commands?

**Comment by MrAdrianR** [comment]
It says:
`fatal: unable to access 'https://github.com/pi-hole/adminLTE.git/': Failed to connect to github.com port 443: No route to host`
both git clone commands.

But I have already unlocked the port.

**Comment by Mcat12** [comment]
Are you able to reach https://github.com ?

**Comment by MrAdrianR** [comment]
To my surprise, I can not load https://github.com in the browser.

I already see that the problem lies with me. Sorry for wasting your time! I will continue to see how I fix the problem.
Until here: Thank you for your support.

**Comment by 91ajames** [comment]
Had this same issue one time and fixed it due to changing pihole a slite bit due to adding speedtest.
Here you go.
https://discourse.pi-hole.net/t/pihole-dashboard-keeps-breaking/6095/2

**Comment by Mcat12** [comment]
@MrAdrianR Have you found a solution?

**Comment by ductator** [comment]
Had the same problem when I wanted to reinstall after FTL bugged out.
https://www.reddit.com/r/pihole/comments/7ldzzt/failed_reconfig_uninstall_reinstall/drml42k/ solved it.

**Comment by MrAdrianR** [comment]
@Mcat12 
Sry i was busy,

I just reinstalled Ubuntu Mate on my Rasp3b, that solved all of my problems.

I did not know what was wrong.

**Comment by lynch3001** [comment]
I came across this exact same issue, but navigated around it by:

 cd /var/www/html/admin/
 sudo git reset --hard

> fatal: Unable to create '/var/www/html/admin/.git/index.lock': File exists.

Another git process seems to be running in this repository, e.g.
an editor opened by 'git commit'. Please make sure all processes
are terminated then try again. If it still fails, a git process
may have crashed in this repository earlier:
remove the file manually to continue.

sudo rm -f ./.git/index.lock
 sudo git reset --hard
sudo git checkout master
sudo git pull
 pihole -up

**Comment by gbroiles** [comment]
I also experienced this issue on Raspbian Stretch Lite (2017-11-29) and Pihole 3.3; the git workaround discussed above didn't help.

Uninstalling lighttpd, dnsmasq, dnsmasq-base, and removing /etc/pihole, /etc/.pihole, and /var/www/html/admin did allow me to reinstall successfully.

**Comment by barrars** [comment]
I think when uninstalling without removing/purging the pihole dependencies seems to contribute to this issue.  I was only able to successfully reinstall after manually removing lighttpd, dnsmasq, dnsmasq-base plus any remaining config paths (ie /etc/pihole, /etc/.pihole, and /var/www/html/admin) at gbroiles suggested.

**Comment by r0ballis0n** [comment]
I was able to get it reinstalled with only removing the /etc/pihole folder. The uninstall had removed the admin folder as well as the .pihole folder. I too did not remove dependencies when I removed it after a failed update to 3.3.

**Comment by jackw01** [comment]
I was able to get it reinstalled by disabling pihole before the update and editing /etc/resolv.conf and changing the DNS server from 127.0.0.1 to 8.8.8.8.

**Comment by EveryOtherUsernameWasAlreadyTaken** [comment]
TLDR: the following worked for me:
```
git clone  https://github.com/pi-hole/pi-hole.git /etc/.pihole 
git clone  git://github.com/pi-hole/AdminLTE.git /var/www/html/admin
```

Details:

```
[i] Clone https://github.com/pi-hole/AdminLTE.git into /var/www/html/admin...
  Error: Could not update local repository. Contact support.
```

I cannot clone the repository using https://

```
# git clone  https://github.com/pi-hole/AdminLTE.git /var/www/html/admin
Cloning into '/var/www/html/admin'...
warning: templates not found /usr/share/git-core/templates
fatal: Unable to find remote helper for 'https'
```

However, it was possible to clone with the git:// version of the repository

```
# git clone  git://github.com/pi-hole/AdminLTE.git /var/www/html/admin
Cloning into '/var/www/html/admin'...
warning: templates not found /usr/share/git-core/templates
remote: Counting objects: 17573, done.
remote: Compressing objects: 100% (111/111), done.
remote: Total 17573 (delta 63), reused 101 (delta 37), pack-reused 17425
Receiving objects: 100% (17573/17573), 24.78 MiB | 1.01 MiB/s, done.
Resolving deltas: 100% (9166/9166), done.
```

**Comment by tyszja** [comment]
I had the exact same issue. If you run the installer without sudo at first try then you're screwed - you can't re-run it with sudo as it will fail updating already cloned repos. You have to delete all of these directories and start over again... with sudo this time.

**Comment by Mcat12** [comment]
The issue has been resolved for the original poster, and other solutions have been given, so this thread can be locked. Please make a new issue if needed.
